src/text-buffer.coffee

```diff
@@ -1539,7 +1539,10 @@ class TextBuffer
   #
   # Returns a {Promise} that resolves when the save has completed.
   save: ->
-    @saveAs(@getPath())
+    if @file?
+      @saveTo(@file)
+    else
+      @saveAs(@getPath())
```

Comment:
I think this method could just do `@saveTo(@file)` with no conditional; it looks like if there is no `@file`, then `@getPath()` will return `null`, so `saveAs` will throw an exception anyway.